File: Algorithms/08_07_2020/Fibonachi/fibonachi.cpp
#include <iostream>
#include <vector>

std::vector<int> vec(2, 1);

int f(int n) {
    if(n < vec.size()) {
        return vec[n];
    }

    int next = f(n - 1) + f(n - 2);
    vec.push_back(next);
    
    return next;
}

int main() {
    std::cout << "Fibonachi 8-th number is: " << f(8) << "\n";
    std::cout << "Fibonachi 10-th number is: " <<f(10) << "\n";

    return 0;
}

File: Python/20_07_2020/fibonachi.py
def fibonachi(n):
    if n==1:
        return 0
    elif n==2:
        return 1
    else:
        return fibonachi(n-1) + fibonachi(n-2)

def main():
    while True:
        number = input("Enter number ")
        print(fibonachi(number))

main()
